File: blockball-api/src/main/java/com/github/shynixn/blockball/api/persistence/entity/TeamMeta.java
package com.github.shynixn.blockball.api.persistence.entity;

import java.util.Optional;

public interface TeamMeta extends Persistenceable<TeamMeta> {
    /**
     * Sets the spawnpoint of the team.
     *
     * @param location location
     */
    void setSpawnPoint(Object location);

    /**
     * Returns the spawnpoint of the team.
     *
     * @return location
     */
    Optional<Object> getSpawnPoint();

    /**
     * Sets the title of the message getting played when a player scores a goal.
     *
     * @param title title
     */
    void setScoreMessageTitle(String title);

    /**
     * Returns the title of the message getting played when a player scores a goal.
     *
     * @return title
     */
    Optional<String> getScoreMessageTitle();

    /**
     * Sets the subTitle of the message getting played when a player scores a goal
     *
     * @param subtitle subtitle
     */
    void setScoreMessageSubtitle(String subtitle);

    /**
     * Returns the subTitle of the message getting played when a player scores a goal
     *
     * @return title
     */
    Optional<String> getScoreMessageSubtitle();

    /**
     * Sets the title of the message getting played when the team wins a match.
     *
     * @param title title
     */
    void setWinningMessageTitle(String title);

    /**
     * Returns the title of the message getting played when the team wins a match.
     *
     * @return title
     */
    Optional<String> getWinningMessageTitle();

    /**
     * Sets the subTitle of the message getting played when the team wins a match.
     *
     * @param subtitle subtitle
     */
    void setWinningMessageSubtitle(String subtitle);

    /**
     * Returns the subTitle of the message getting played when the team wins a match.
     *
     * @return title
     */
    Optional<String> getWinningMessageSubtitle();

    /**
     * Sets the name of the team which gets displayed by the placeholder :red or :blue.
     *
     * @param name name
     */
    void setDisplayName(String name);

    /**
     * Returns he name of the team which gets displayed by the placeholder :red or :blue.
     *
     * @return name
     */
    String getDisplayName();

    /**
     * Sets the prefix of the team which gets displayed by the placeholder :redcolor or :bluecolor.
     *
     * @param prefix prefix
     */
    void setPrefix(String prefix);

    /**
     * Returns the prefix of the team which gets displayed by the placeholder :redcolor or :bluecolor.
     *
     * @return prefix
     */
    String getPrefix();

    /**
     * Returns the armor items of the team.
     *
     * @return armor
     */
    Object[] getArmorContents();

    /**
     * Sets the armor items of the team.
     *
     * @param itemStacks armor
     */
    void setArmorContents(Object[] itemStacks);
}

File: blockball-api/src/main/java/com/github/shynixn/blockball/api/persistence/entity/TeamMeta.kt
package com.github.shynixn.blockball.api.persistence.entity

interface TeamMeta {

    /** Goal properties of the team. */
    val goal: Selection

    /** Spawnpoint of the team inside of the arena. */
    var spawnpoint: Position?

    /** Optional lobby spawnpoint */
    var lobbySpawnpoint: Position?

    /** DisplayName of the team which gets used in the placeholder <red> or <blue>. */
    var displayName: String

    /** Prefix of the team which gets used in the placeholder <redcolor> or <bluecolor>. */
    var prefix: String

    /** Min amount of players in this team to start the match for this team. */
    var minAmount: Int

    /** Max amount of players in this team to start the match for this team. */
    var maxAmount: Int
    /** Minimum amount of players in this team to keep the game running. **/
    var minPlayingPlayers : Int

    /** Amount of points this team receives when a goal gets scored. */
    var pointsPerGoal: Int

    /** Amount of points this team receives when a player of the opposite team dies. */
    var pointsPerEnemyDeath: Int

    /** WalkingSpeed of the players in this team. */
    var walkingSpeed: Double

    /** Armor wearing this team. */
    var armorContents: Array<Any?>

    /** Inventory this team is getting when playing. */
    var inventoryContents: Array<Any?>

    /** Title of the message getting played when a player scores a goal. */
    var scoreMessageTitle: String

    /** Subtitle of the message getting played when a player scores a goal. */
    var scoreMessageSubTitle: String

    /** Title of the message getting played when this team wins a match. */
    var winMessageTitle: String

    /** Subtitle of the message getting played when this team wins a match. */
    var winMessageSubTitle: String

    /** Title of the message getting played when the match ends in a draw.*/
    var drawMessageTitle: String

    /** Subtitle of the message getting played when the match ends in a draw. */
    var drawMessageSubTitle: String

    /** Message getting played when a player joins a match.*/
    var joinMessage: String

    /** Message getting played when a player leave a match.*/
    var leaveMessage: String

    /** Lines displayed on the sign for joining the team. */
    var signLines: List<String>

    /** List of signs which can be clicked to join the red team.*/
    val signs: MutableList<Position>
}
